### PR TITLE
[NeMo-UX] Adding recipes

### PR DESCRIPTION
**Pull request by marcromeyn**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/llm/api.py [hunk 1]
```diff
@@ -10,7 +10,6 @@
 from nemo.lightning.pytorch.callbacks import PEFT, ModelTransform
 from nemo.utils import logging
 
-
```
Comment (akoumpa) [inline nemo/collections/llm/api.py:13]: if that's a git move can we drop the line removal?

File: nemo/collections/llm/models/llama3_8b_64k.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    return LlamaModel(Llama3Config8B(seq_length=65536))
+
+
+@factory(name=NAME)
+def trainer(devices=8) -> nl.Trainer:
+    strategy = nl.MegatronStrategy(
+        tensor_model_parallel_size=8,
+        context_parallel_size=4,
+        sequence_parallel=True,
+        expert_model_parallel_size=1,
```
Comment (akoumpa) [inline nemo/collections/llm/models/llama3_8b_64k.py:25]: is this line needed?

**Review by akoumpa — CHANGES_REQUESTED** [review]
Can we rename the directory `models` to `recipes`?

Otherwise it's confusing why there's a mistral.py under both `collections/llm/gpt/model` and `collections/llm/models`. 
I don't have any strong preference as long as one of the two directories changes name.

File: nemo/collections/llm/models/llama3_8b_64k.py (hunk 2 above)
Comment (ashors1) [reply nemo/collections/llm/models/llama3_8b_64k.py]: No, I removed it

**Comment by ashors1** [comment]
> Can we rename the directory `models` to `recipes`?
> 
> Otherwise it's confusing why there's a mistral.py under both `collections/llm/gpt/model` and `collections/llm/models`. I don't have any strong preference as long as one of the two directories changes name.

Done

**Review by akoumpa — APPROVED** [review]

**Pull request merged**